README.md
# x-eth-tool

Suite per gestione unità di rete

# Installazione:
decomprimere il file nel persorso<br>
	/home/USER/cmd/<br>
con il comando:<br>
	tar -zxvf x-eth-tool-tar.gz<br>
<br>
Installare, poi, i vari moduli. Es.<br>
	./ctrlx_0.3.sh -i

# Configurazione
dopo aver installato tutti i demoni, configuriamo il file dei parametri, prima lo rinominiamo con:<br>
	mv parametri-server-nuova.conf parametri-server.conf<br>
poi lo modifichiamo per aggiungere i server:<br>
	nano parametri-server.conf


changelog.txt
10/11/2017
  * Aggiornato CTRLX; rimosso l'errore che restituiva quando non si inseriva il nome dell'applicazione e si voleva uscire.

14-08-2017:
  * Aggiornati Connectx e cpserver con estensione dei server memorizzati, da adesso sono 10

2/10/2016:
  * Aggiunto Script per smontaggio unità

27-09-2016:
  * Estensione server memorizzati, da adesso sono 5

26-09-2016:
  * Aggiunta possibilità di interrogare server non memorizzati, quick services

25-09-16
  * Correzione bug

19-09-16:
  * Aggiunto script util_script.sh

16-09-2016:
  * Aggiunto lo script cp_server.sh
  * Risoluzioni di bug e migliorie

22-07-16:
  * Script reso multi server

06-06-2016:
  * Risoluzioni di bug e migliorie

11-04-2016:
  * Rilascio versione iniziale


mount_x.sh
#!/bin/bash
#### BEGIN INIT INFO
# Info script e set variabili
  Author="Giuseppe Mangiameli @ IT9ARZ"
# Data di rilascio: 20/09/2016 
  Date="27/09/2016" #Revision
  Version="1.0.5"
  TitleScript="mountx"
# Description: Connetti e monta unità di rete
# Required: Credenziali presenti in parametri-server.conf
#           richiesto script_util.sh
#### END INIT INFO

#carico i parametri
source ~/cmd/parametri-server.conf
source ~/cmd/script_util.sh

# clear the screen
clear

# Define Variable acmreset
acmreset=$(tput sgr0)

while getopts ivr name
do
        case $name in
          i)iopt=1;;
          v)vopt=1;;
          r)ropt=1;;
          *)echo "Argomento errato";;
        esac
done

if [[ ! -z $iopt ]]
then
{
wd=$(pwd)
basename "$(test -L "$0" && readlink "$0" || echo "$0")" > /tmp/scriptname
scriptname=$(echo -e -n $wd/ && cat /tmp/scriptname)
echo "Installazione in corso di $TitleScript"
su -c "cp $scriptname /usr/bin/$TitleScript" root && e_success "Script Installato! Esegui ora il Comando $TitleScript" || e_error "!! Errore nella fase di Installazione di $TitleScript"
}
fi

if [[ ! -z $vopt ]]
then
{
e_header_1 "$TitleScript versione: $Version   Revision: $Date"
e_txt "Designed by $Author"
e_txt "Monta Unità di rete sul sistema"
e_warning "Inserire Argomento "-i" per installare lo script"
e_warning "Inserire Argomento "-v" per vedere questa info"
e_warning "Inserire Argomento "-r" per rimuovere lo script"
e_warning "Si richiede l'installazione di <sshfs> - # apt-get install sshfs fuse "
}
fi

if [[ ! -z $ropt ]]
then
{
su -c "rm /usr/bin/$TitleScript" root && e_success "$TitleScript rimosso con successo!" || e_error "!! Errore nella rimozione di $TitleScript"
}
fi

if [[ $# -eq 0 ]]
then
{
#domando il tipo di server da interrogare
echo -e "                              _         "
echo -e "                             | |        "
echo -e "  _ __ ___   ___  _   _ _ __ | |___  __ "
echo -e " | '_ ' _ \ / _ \| | | | '_ \| __\ \/ / "
echo -e " | | | | | | (_) | |_| | | | | |_ >  <  "
echo -e " |_| |_| |_|\___/ \__,_|_| |_|\__/_/\_\ "
echo "by $Author"                                         
e_header "montaggio unità di rete"
e_txt "Scegli il server con l'unità da montare."
e_txt "per annullare il comando, lasciare vuoto e premere invio."
e_warning "z. Inserisci Server"
e_warning "$es1"
e_warning "$es2"
e_warning "$es3"
e_warning "$es4"
e_warning "$es5"
echo -n "Scegli: "

#Recupero la risposta
read ES

if [ ${ES:-y} = "a" ];
then
{
SERVER=$a
}
elif [ ${ES:-y} = "b" ];
then
{
SERVER=$b
}
elif [ ${ES:-y} = "c" ];
then
{
SERVER=$c
}
elif [ ${ES:-y} = "d" ];
then
{
SERVER=$d
}
elif [ ${ES:-y} = "e" ];
then
{
SERVER=$e
}
elif [ ${ES:-y} = "z" ];
then
{
echo -n "Inserisci Nome Utente: "
#Recupero la risposta
read Z_Utente
#echo -n "Inserisci Password: "
#Recupero la risposta
#read Z_Passwd
echo -n "Inserisci Host: "
#Recupero la risposta
read Z_Host
echo -n "Inserisci Porta: "
#Recupero la risposta
read Z_Porta
echo -n "Inserisci Posizione di montaggio: "
#Recupero la risposta
read Z_pos
echo -n "Inserisci Posizione Remota: "
#Recupero la risposta
read Z_root
#Avvio Connessione verso Server
sshfs $Z_Utente@$Z_Host:$Z_root $Z_pos -p $Z_Porta
exit
}
elif [ ${ES:-y} = "y" ];
then
{
e_error "comando annullato"
exit
}
else
{
   e_error "Comando errato  -  $USER devi scegliere i server in elenco."
   exit
}
fi

#domando il tipo di connessione da stabilire
e_txt "Scegli il tipo di connessione per connetterti al Server: <"$SERVER">."
e_txt "per annullare il comando, lasciare vuoto e premere invio."
e_warning "LAN <l>"
e_warning "WAN <w>"
echo -n "scegli: "

#Recupero la risposta
read CONNECT

#Connessione tipo LAN
if [ ${CONNECT:-x} = "l" ];
then
{
#Informo il tipo di connessione scelta
e_success "inizio connessione SSH verso <"$SERVER"> su rete LAN."

if [ ${ES:-x} = "a" ];
then
{
sshfs $A_user@$A_lhost:$A_root $A_pos -p $A_ldoor
}

elif [ ${ES:-x} = "b" ];
then
{
sshfs $B_user@$B_lhost:$B_root $B_pos -p $B_ldoor
}

elif [ ${ES:-x} = "c" ];
then
{
sshfs $C_user@$C_lhost:$C_root $C_pos -p $C_ldoor
}

elif [ ${ES:-x} = "d" ];
then
{
sshfs $D_user@$D_lhost:$D_root $D_pos -p $D_ldoor
}

elif [ ${ES:-x} = "e" ];
then
{
sshfs $E_user@$E_lhost:$E_root $E_pos -p $E_ldoor
}
fi
}
fi


#Connessione tipo WAN
if [ ${CONNECT:-x} = "w" ];
then
{
#Informo il tipo di connessione scelta
e_success "inizio connessione SSH verso <"$SERVER"> su rete WAN"

if [ ${ES:-x} = "a" ];
then
{
sshfs $A_user@$A_rhost:$A_root $A_pos -p $A_door
}

elif [ ${ES:-x} = "b" ];
then
{
sshfs $B_user@$B_rhost:$B_root $B_pos -p $B_door
}

elif [ ${ES:-x} = "c" ];
then
{
sshfs $C_user@$C_rhost:$C_root $C_pos -p $C_door
}

elif [ ${ES:-x} = "d" ];
then
{
sshfs $D_user@$D_rhost:$D_root $D_pos -p $D_door
}

elif [ ${ES:-x} = "e" ];
then
{
sshfs $E_user@$E_rhost:$E_root $E_pos -p $E_door
}
fi
}
fi

#Comando annullato
if [ ${CONNECT:-x} = "x" ];
then
	e_error "comando annullato"
fi

}
fi

shift $(($OPTIND -1))

connect_x.sh
#!/bin/bash
#### BEGIN INIT INFO
# Info script e set variabili
  Author="Giuseppe Mangiameli @ IT9ARZ"
# Data di rilascio: 11/04/2016  
  Date="27/09/2016" #Revision
  Version="2.2.4"
  TitleScript="connectx"
# Description: Avvia connessione sicura verso Server via SSH
# Required: Credenziali presenti in parametri-server.conf
#           richiesto script_util.sh
#### END INIT INFO

#carico i parametri
source ~/cmd/parametri-server.conf
source ~/cmd/script_util.sh

# clear the screen
clear

# Define Variable acmreset
acmreset=$(tput sgr0)

while getopts ivr name
do
        case $name in
          i)iopt=1;;
          v)vopt=1;;
          r)ropt=1;;
          *)echo "Argomento errato";;
        esac
done

if [[ ! -z $iopt ]]
then
{
wd=$(pwd)
basename "$(test -L "$0" && readlink "$0" || echo "$0")" > /tmp/scriptname
scriptname=$(echo -e -n $wd/ && cat /tmp/scriptname)
echo "Installazione in corso di $TitleScript"
su -c "cp $scriptname /usr/bin/$TitleScript" root && e_success "Script Installato! Esegui ora il Comando $TitleScript" || e_error "!! Errore nella fase di Installazione di $TitleScript"
}
fi

if [[ ! -z $vopt ]]
then
{
e_header_1 "$TitleScript versione: $Version   Revision: $Date"
e_txt "Designed by $Author"
e_txt "Consente di collegarci in SSH ad un server"
e_warning "Inserire Argomento "-i" per installare lo script"
e_warning "Inserire Argomento "-v" per vedere questa info"
e_warning "Inserire Argomento "-r" per rimuovere lo script"
}
fi

if [[ ! -z $ropt ]]
then
{
su -c "rm /usr/bin/$TitleScript" root && e_success "$TitleScript rimosso con successo!" || e_error "!! Errore nella rimozione di $TitleScript"
}
fi

if [[ $# -eq 0 ]]
then
{

#domando il tipo di server da interrogare
echo "                                 _         "
echo "                                | |        "
echo "  ___ ___  _ __  _ __   ___  ___| |___  __ "
echo " / __/ _ \| '_ \| '_ \ / _ \/ __| __\ \/ / "
echo "| (_| (_) | | | | | | |  __/ (__| |_ >  <  "
echo " \___\___/|_| |_|_| |_|\___|\___|\__/_/\_\ "
echo "by $Author"                                        
e_header "Connessioni SSH Server"
e_txt "Scegliere su quale server collegarti."
e_txt "per annullare il comando, lasciare vuoto e premere invio."
e_warning "z. Inserisci Server"
e_warning "$es1"
e_warning "$es2"
e_warning "$es3"
e_warning "$es4"
e_warning "$es5"
echo -n "Scegli: "

#Recupero la risposta
read ES

if [ ${ES:-y} = "a" ];
then
{
SERVER=$a
}
elif [ ${ES:-y} = "b" ];
then
{
SERVER=$b
}
elif [ ${ES:-y} = "c" ];
then
{
SERVER=$c
}
elif [ ${ES:-y} = "d" ];
then
{
SERVER=$d
}
elif [ ${ES:-y} = "e" ];
then
{
SERVER=$e
}
elif [ ${ES:-y} = "z" ];
then
{
echo -n "Inserisci Nome Utente: "
#Recupero la risposta
read Z_Utente
#echo -n "Inserisci Password: "
#Recupero la risposta
#read Z_Passwd
echo -n "Inserisci Host: "
#Recupero la risposta
read Z_Host
echo -n "Inserisci Porta: "
#Recupero la risposta
read Z_Porta
#Avvio Connessione verso Server
ssh $Z_Utente@$Z_Host -p $Z_Porta
exit
}
elif [ ${ES:-y} = "y" ];
then
{
e_error "comando annullato"
exit
}
else
{
   e_error "Comando errato  -  $USER devi scegliere i server in elenco."
   exit
}
fi

#domando il tipo di connessione da stabilire
e_txt "Scegli il tipo di connessione per connetterti al Server: <"$SERVER">."
e_txt "per annullare il comando, lasciare vuoto e premere invio."
e_warning "LAN <l>"
e_warning "WAN <w>"
echo -n "scegli: "

#Recupero la risposta
read CONNECT

#Connessione tipo LAN
if [ ${CONNECT:-x} = "l" ];
then
{
#Informo il tipo di connessione scelta
e_success "inizio connessione SSH verso <"$SERVER"> su rete LAN."

if [ ${ES:-x} = "a" ];
then
{
ssh $A_user@$A_lhost -p $A_ldoor
}

elif [ ${ES:-x} = "b" ];
then
{
ssh $B_user@$B_lhost -p $B_ldoor
}

elif [ ${ES:-x} = "c" ];
then
{
ssh $C_user@$C_lhost -p $C_ldoor
}

elif [ ${ES:-x} = "d" ];
then
{
ssh $D_user@$D_lhost -p $D_ldoor
}
elif [ ${ES:-x} = "e" ];
then
{
ssh $E_user@$E_lhost -p $E_ldoor
}
fi
}
fi


#Connessione tipo WAN
if [ ${CONNECT:-x} = "w" ];
then
{
#Informo il tipo di connessione scelta
e_success "inizio connessione SSH verso <"$SERVER"> su rete WAN"

if [ ${ES:-x} = "a" ];
then
{
ssh $A_user@$A_rhost -p $A_door
}

elif [ ${ES:-x} = "b" ];
then
{
ssh $B_user@$B_rhost -p $B_door
}

elif [ ${ES:-x} = "c" ];
then
{
ssh $C_user@$C_rhost -p $C_door
}

elif [ ${ES:-x} = "d" ];
then
{
ssh $D_user@$D_rhost -p $D_door
}

elif [ ${ES:-x} = "e" ];
then
{
ssh $E_user@$E_rhost -p $E_door
}
fi
}
fi

#Comando annullato
if [ ${CONNECT:-x} = "x" ];
then
	e_error "comando annullato"
fi

}
fi

shift $(($OPTIND -1))

script_util.sh
#!/bin/bash
#### BEGIN INIT INFO
# Author: Giuseppe Mangiameli @ IT9ARZ
# Date: 19/09/2016
# Version 0.1
# Description: Setto alcune variabili utili per gli script
#### END INIT INFO

#
# Set Colors
#

bold=$(tput bold)
underline=$(tput sgr 0 1)
reset=$(tput sgr0)

purple=$(tput setaf 171)
red=$(tput setaf 1)
green=$(tput setaf 76)
tan=$(tput setaf 3)
blue=$(tput setaf 38)

#
# Headers and  Logging
#

e_header() { printf "\n${bold}${purple}==========  %s  ==========${reset}\n" "$@"
}
e_header_1() { printf "\n${bold}${tan}#####  %s  #####${reset}\n" "$@"
}
e_arrow() { printf "➜ $@\n"
}
e_success() { printf "${green}✔ %s${reset}\n" "$@"
}
e_error() { printf "${red}✖ %s${reset}\n" "$@"
}
e_warning() { printf "${tan}➜ %s${reset}\n" "$@"
}
e_underline() { printf "${underline}${bold}%s${reset}\n" "$@"
}
e_bold() { printf "${bold}%s${reset}\n" "$@"
}
e_note() { printf "${underline}${bold}${blue}Note:${reset}  ${blue}%s${reset}\n" "$@"
}
e_txt() { printf "${blue}%s${reset}\n" "$@"
}

#
# USAGE FOR SEEKING CONFIRMATION
# seek_confirmation "Ask a question"
# Credt: https://github.com/kevva/dotfiles
#
# if is_confirmed; then
#   some action
# else
#   some other action
# fi
#

seek_confirmation() {
  printf "\n${bold}$@${reset}"
  read -p " (y/n) " -n 1
  printf "\n"
}

# underlined
seek_confirmation_head() {
  printf "\n${underline}${bold}$@${reset}"
  read -p "${underline}${bold} (y/n)${reset} " -n 1
  printf "\n"
}

# Test whether the result of an 'ask' is a confirmation
is_confirmed() {
if [[ "$REPLY" =~ ^[Yy]$ ]]; then
  return 0
fi
return 1
}

cp_server.sh
#!/bin/bash
#### BEGIN INIT INFO
# Info script e set variabili
  Author="Giuseppe Mangiameli @ IT9ARZ"
# Data di rilascio: 16/09/2016 
  Date="27/09/2016" #Revision
  Version="2.2.4"
  TitleScript="cpserver"
# Description: Copia un file da/verso Server a sistema Locale. Possibile scegliere se LAN o WAN
# Required:	Credenziali presenti in parametri-server.conf
#           richiesto script_util.sh
#### END INIT INFO

#carico i parametri
source ~/cmd/parametri-server.conf
source ~/cmd/script_util.sh

# clear the screen
clear

# Define Variable acmreset
acmreset=$(tput sgr0)

while getopts ivr name
do
        case $name in
          i)iopt=1;;
          v)vopt=1;;
          r)ropt=1;;
          *)echo "Argomento errato";;
        esac
done

if [[ ! -z $iopt ]]
then
{
wd=$(pwd)
basename "$(test -L "$0" && readlink "$0" || echo "$0")" > /tmp/scriptname
scriptname=$(echo -e -n $wd/ && cat /tmp/scriptname)
echo "Installazione in corso di $TitleScript"
su -c "cp $scriptname /usr/bin/$TitleScript" root && e_success "Script Installato, esegui ora il Comando $TitleScript" || e_error "Errore nella fase di Installazione di $TitleScript"
}
fi

if [[ ! -z $vopt ]]
then
{
e_header_1 "$TitleScript versione: $Version   Revision: $Date"
e_txt "Designed by $Author"
e_txt "copia files da server e verso server"
e_warning "Inserire Argomento "-i" per installare lo script"
e_warning "Inserire Argomento "-v" per vedere questa info"
e_warning "Inserire Argomento "-r" per rimuovere lo script"
}
fi

if [[ ! -z $ropt ]]
then
{
su -c "rm /usr/bin/$TitleScript" root && e_success "$TitleScript rimosso con successo" || e_error "Errore nella rimozione di $TitleScript"
}
fi

if [[ $# -eq 0 ]]
then
{

#domando il tipo di server da interrogare                                          
echo "   ___ _ __  ___  ___ _ ____   _____ _ __  "
echo "  / __| '_ \/ __|/ _ \ '__\ \ / / _ \ '__| "
echo " | (__| |_) \__ \  __/ |   \ V /  __/ |    "
echo "  \___| .__/|___/\___|_|    \_/ \___|_|    "
echo "      | |                                  "
echo "      |_|                                  "
echo "by $Author"      
e_header "trasferimenti files SSH su server"
e_txt "Scegliere su quale server collegarti per iniziare il trasferimento di dati"
e_txt "per annullare il comando, lasciare vuoto e premere invio"
e_warning "z. Inserisci Server"
e_warning "$es1"
e_warning "$es2"
e_warning "$es3"
e_warning "$es4"
e_warning "$es5"
echo -n "Scegli: "

#Recupero la risposta
read ES

if [ ${ES:-y} = "a" ];
then
{
SERVER=$a
}
elif [ ${ES:-y} = "b" ];
then
{
SERVER=$b
}
elif [ ${ES:-y} = "c" ];
then
{
SERVER=$c
}
elif [ ${ES:-y} = "d" ];
then
{
SERVER=$d
}
elif [ ${ES:-y} = "e" ];
then
{
SERVER=$e
}
elif [ ${ES:-y} = "z" ];
then
{
e_txt "scegli se copiare file da:"
e_warning "Locale > Remoto <r>"
e_warning "Remoto > Locale <l>"
echo -n "scegli: "

#Recupero la risposta
read Z_Risposta

if [ ${Z_Risposta:-u} = "l" ];
then
{
echo -n "Inserisci Nome Utente: "
#Recupero la risposta
read Z_Utente
#echo -n "Inserisci Password: "
#Recupero la risposta
#read Z_Passwd
echo -n "Inserisci Host: "
#Recupero la risposta
read Z_Host
echo -n "Inserisci Porta: "
#Recupero la risposta
read Z_Porta
echo -n "Inserisci Posizione Locale: "
#Recupero la risposta
read Z_POS_LOCALE
echo -n "Inserisci Posizione Remota: "
#Recupero la risposta
read Z_POS_REMOTA
#Avvio Connessione verso Server
scp -P $Z_Porta -rpC $Z_Utente@$Z_Host:$Z_POS_REMOTA $Z_POS_LOCALE
exit
}
fi

if [ ${Z_Risposta:-u} = "r" ];
then
{
echo -n "Inserisci Nome Utente: "
#Recupero la risposta
read Z_Utente
#echo -n "Inserisci Password: "
#Recupero la risposta
#read Z_Passwd
echo -n "Inserisci Host: "
#Recupero la risposta
read Z_Host
echo -n "Inserisci Porta: "
#Recupero la risposta
read Z_Porta
echo -n "Inserisci Posizione Locale: "
#Recupero la risposta
read Z_POS_LOCALE
echo -n "Inserisci Posizione Remota: "
#Recupero la risposta
read Z_POS_REMOTA
#Avvio Connessione verso Server
scp -P $Z_Porta -rpC $Z_POS_LOCALE $Z_Utente@$Z_Host:$Z_POS_REMOTA
exit
}
fi

#Comando annullato
if [ ${Z_Risposta:-u} = "u" ];
then
{
	e_error "comando annullato"
	exit
}
else
{
   e_error "Comando errato  -  $USER devi scrivere l oppure r"
   exit
}
fi
}


elif [ ${ES:-y} = "y" ];
then
{
e_error "comando annullato"
exit
}
else
{
   e_error "Comando errato  -  $USER devi scegliere i server in elenco"
   exit
}
fi

#domando il tipo di connessione da stabilire
e_txt "Scegli il tipo di connessione per connetterti al Server: <"$SERVER">"
e_txt "per annullare il comando, lasciare vuoto e premere invio"
e_warning "LAN <l>"
e_warning "WAN <w>"
echo -n "scegli: "

#Recupero la risposta
read CONNECT

#Connessione tipo LAN
if [ ${CONNECT:-x} = "l" ];
then
{
#domando il tipo di connessione da stabilire
e_success "inizio connessione SSH verso <"$SERVER"> su rete LAN"
e_txt "per annullare il comando, lasciare vuoto e premere invio"
e_txt "scegli se copiare file da:"
e_warning "Locale > Remoto <r>"
e_warning "Remoto > Locale <l>"
echo -n "scegli: "

#Recupero la risposta
read RISPOSTA

#converto la risposta digitata in un indirizzo host corretto
#":-"x viene usata come risposta di default, in questo caso sarebbe "x"
if [ ${RISPOSTA:-x} = "l" ];
then
{
#domando la posizione remota del file
e_txt "Copia File da Server <"$SERVER"> > verso questo Sistema"
echo -n "Digita la posizione remota del file: "

#Recupero la risposta
read POS_REMOTA

#domando la posizione locale dove si intendere copiare il file
echo -n "Digita la posizione locale: "

#Recupero la risposta
read POS_LOCALE

if [ ${ES:-x} = "a" ];
then
{
scp -P $A_ldoor -rpC $A_user@$A_lhost:$POS_REMOTA $POS_LOCALE
}

elif [ ${ES:-x} = "b" ];
then
{
scp -P $B_ldoor -rpC $B_user@$B_lhost:$POS_REMOTA $POS_LOCALE
}

elif [ ${ES:-x} = "c" ];
then
{
scp -P $C_ldoor -rpC $C_user@$C_lhost:$POS_REMOTA $POS_LOCALE
}

elif [ ${ES:-x} = "d" ];
then
{
scp -P $D_ldoor -rpC $D_user@$D_lhost:$POS_REMOTA $POS_LOCALE
}

elif [ ${ES:-x} = "e" ];
then
{
scp -P $E_ldoor -rpC $E_user@$E_lhost:$POS_REMOTA $POS_LOCALE
}
fi
}
fi

if [ ${RISPOSTA:-x} = "r" ];
then
{
#domando la posizione remota del file
e_txt "Copia File da questo Sistema verso > Server <"$SERVER">"
echo -n "Digita la posizione locale del file da copiare: "

#Recupero la risposta
read POS_LOCALE

#domando la posizione locale dove si intendere copiare il file
echo -n "Digita la posizione remota dove salvare il file: "

#Recupero la risposta
read POS_REMOTA

if [ ${ES:-x} = "a" ];
then
{
scp -P $A_ldoor -rpC $POS_LOCALE $A_user@$A_lhost:$POS_REMOTA
}

elif [ ${ES:-x} = "b" ];
then
{
scp -P $B_ldoor -rpC $POS_LOCALE $B_user@$B_lhost:$POS_REMOTA
}

elif [ ${ES:-x} = "c" ];
then
{
scp -P $C_ldoor -rpC $POS_LOCALE $C_user@$C_lhost:$POS_REMOTA
}

elif [ ${ES:-x} = "d" ];
then
{
scp -P $D_ldoor -rpC $POS_LOCALE $D_user@$D_lhost:$POS_REMOTA
}

elif [ ${ES:-x} = "e" ];
then
{
scp -P $E_ldoor -rpC $POS_LOCALE $E_user@$E_lhost:$POS_REMOTA
}
fi
}
fi

#Comando annullato
if [ ${RISPOSTA:-x} = "x" ];
then
{
	e_error "comando annullato"
	exit
}
else
{
   e_error "Comando errato  -  $USER devi scrivere l oppure r"
   exit
}
fi

}
fi

#Connessione tipo WAN
if [ ${CONNECT:-x} = "w" ];
then
{
#domando il tipo di connessione da stabilire
e_success "inizio connessione SSH verso <"$SERVER"> su rete WAN"
e_txt "per annullare il comando, lasciare vuoto e premere invio"
e_txt "scegli se copiare file da:"
e_warning "Locale > Remoto <r>"
e_warning "Remoto > Locale <l>"
echo -n "scegli: "

#Recupero la risposta
read RISPOSTA

#converto la risposta digitata in un indirizzo host corretto
#":-"x viene usata come risposta di default, in questo caso sarebbe "x"
if [ ${RISPOSTA:-x} = "l" ];
then
{
#domando la posizione remota del file
e_txt "Copia File da Server <"$SERVER"> > verso questo Sistema"
echo -n "Digita la posizione remota del file: "

#Recupero la risposta
read POS_REMOTA

#domando la posizione locale dove si intendere copiare il file
echo -n "Digita la posizione locale: "

#Recupero la risposta
read POS_LOCALE

if [ ${ES:-x} = "a" ];
then
{
scp -P $A_door -rpC $A_user@$A_rhost:$POS_REMOTA $POS_LOCALE
}

elif [ ${ES:-x} = "b" ];
then
{
scp -P $B_door -rpC $B_user@$B_rhost:$POS_REMOTA $POS_LOCALE
}

elif [ ${ES:-x} = "c" ];
then
{
scp -P $C_door -rpC $C_user@$C_rhost:$POS_REMOTA $POS_LOCALE
}

elif [ ${ES:-x} = "d" ];
then
{
scp -P $D_door -rpC $D_user@$D_rhost:$POS_REMOTA $POS_LOCALE
}

elif [ ${ES:-x} = "e" ];
then
{
scp -P $E_door -rpC $E_user@$E_rhost:$POS_REMOTA $POS_LOCALE
}
fi
}
fi

if [ ${RISPOSTA:-x} = "r" ];
then
{
#domando la posizione remota del file
e_txt "Copia File da questo Sistema verso > Server <"$SERVER">"
echo -n "Digita la posizione locale del file: "

#Recupero la risposta
read POS_LOCALE

#domando la posizione locale dove si intendere copiare il file
echo -n "Digita la posizione remota dove salvare il file: "

#Recupero la risposta
read POS_REMOTA

if [ ${ES:-x} = "a" ];
then
{
scp -P $A_door -rpC $POS_LOCALE $A_user@$A_rhost:$POS_REMOTA
}

elif [ ${ES:-x} = "b" ];
then
{
scp -P $B_door -rpC $POS_LOCALE $B_user@$B_rhost:$POS_REMOTA
}

elif [ ${ES:-x} = "c" ];
then
{
scp -P $C_door -rpC $POS_LOCALE $C_user@$C_rhost:$POS_REMOTA
}

elif [ ${ES:-x} = "d" ];
then
{
scp -P $D_door -rpC $POS_LOCALE $D_user@$D_rhost:$POS_REMOTA
}

elif [ ${ES:-x} = "e" ];
then
{
scp -P $E_door -rpC $POS_LOCALE $E_user@$E_rhost:$POS_REMOTA
}
fi
}
fi

#Comando annullato
if [ ${RISPOSTA:-x} = "x" ];
then
{
	e_error "comando annullato"
	exit
}
else
{
   e_error "Comando errato  -  $USER devi scrivere l oppure r"
   exit
}
fi

}
fi

#Comando annullato
if [ ${CONNECT:-x} = "x" ];
then
{
	e_error "comando annullato"
	exit
}
else
{
   e_error "Comando errato  -  $USER devi scrivere l oppure w"
   exit
}
fi
}
fi
shift $(($OPTIND -1))
